### ISSUE TITLE
Test after package build fails on Windows if whitespace in root path

### ISSUE THREAD

**Issue by wulmer** [conda/conda-build#1444]
Hi,
when I install Anaconda to `C:\Program Files\Anaconda3` (other paths not possible, see https://github.com/ContinuumIO/anaconda-issues/issues/1147), it seems that `conda build` doesn't deal properly with the whitespace in the path.
Package build works, but in the test step, I get the following output:

    TEST START: packagexy-0.1.0-py35_0
    [...]
    The following packages will be downloaded:

        package                    |            build
       ---------------------------|-----------------
       packagexy-0.1.0            |           py35_0           2 KB  local

    The following NEW packages will be INSTALLED:
    [...]

    C:\Users\XXXX\conda-bld\test_pkg_1476116784137\test_tmp>call C:\Program Files\Anaconda3\Scripts\activate.bat C:\Users\XXXX\conda-bld\test_pkg_1476116784137\_t_env  1>nul 2>&1

    C:\Users\XXXX\conda-bld\test_pkg_1476116784137\test_tmp>if errorlevel 1 exit 1

    TESTS FAILED: packagexy-0.1.0-py35_0

If you look at the line with the `call C:\...`, you will notice that this line will fail due to the whitespace in the path to activate.bat. CMD will read `call C:\Program [some other args]` which is invalid as `C:\Program` doesn't exist.

Adding quotes around the path to `activate.bat` will resolve this issue.

Using conda-build 2.0.2 and conda 4.2.9 on Windows 7 with Anaconda 3 64bit.

**Comment by msarahan** [comment]
Thanks.  Will fix for next release.  Workaround might be to pass ```--no-activate``` to conda build.  This does not activate the environment, but does prepend the appropriate paths.

**Comment by wulmer** [comment]
Thanks for the suggested workaround. Unfortunately it only brings up another error: Running `conda build --no-activate packagexy` ends also in the test phase with:

    TEST START: packagexy-0.1.0-py35_0
    updating index in: C:\Users\XXXXXX\conda-bld\win-64
    updating index in: C:\Users\XXXXXX\conda-bld\noarch
    
    The following packages will be downloaded:
    
        package                    |            build
        ---------------------------|-----------------
        packagexy-0.1.0            |           py35_0           2 KB  local
    
    The following NEW packages will be INSTALLED:
    
    [...]
    
    WARNING conda.lock:touch(53): Failed to create lock, do not run conda in paralle
    l processes [errno 13]
    Runtime error: Could not open 'C:\\Program Files\\Anaconda3\\pkgs\\packagexy-0.1
    .0-py35_0.tar.bz2.part' for writing ([Errno 13] Permission denied: 'C:\\Program
    Files\\Anaconda3\\pkgs\\packagexy-0.1.0-py35_0.tar.bz2.part').

I have seen this error before, see [this thread on the conda mailing list](https://groups.google.com/a/continuum.io/forum/#!topic/conda/fMqO2GNQqSU)

**Comment by wulmer** [comment]
Note on the previous comment: Running `conda build --no-activate packagexy` doesn't work with conda-build 2.0.2 but works with conda-build 2.0.4.

So it seems to be already fixed. However this doesn't help me as I need a full Anaconda installer which contains this fix for conda-build. 

@msarahan Will there be an Anaconda release 4.2.1 soon?

**Comment by msarahan** [comment]
We do not have current plans for a 4.2.1 Anaconda release.  The next general Anaconda release is planned for the end of January.  If you need a custom installer sooner, I recommend Constructor: https://github.com/conda/constructor which should allow you to create your own installers with similar features to Anaconda, but with whatever packages (and package versions) you want.

**Comment by wulmer** [comment]
Ok, thanks. I gave *constructor* another try (after I tried it a few months ago) and it looks very promising. I noticed a few differences in the installed `_nsis.py` module, but everything else works. Great work!

I will close this issue here as the original problem with the missing file permissions is already addressed in a number of conda issues, e.g.  conda/conda#3053, conda/conda#3448, conda/conda#3469, conda/conda#3604 and others.
